src/global.d.ts

```diff
@@ -5,3 +5,7 @@ interface Window {
 interface Clipboard extends EventTarget {
   write(data: any[]): Promise<void>;
 }
+
+type Mutable<T> = {
+  -readonly [P in keyof T]: T[P];
```

Comment:
TIL. this is awesome! i was really struggling with how to implement this and had no idea this typescript feature existed.

does this belong in global.d.ts or would we rather import it from somewhere? i don't have a strong preference either way

Reply:
- ad `does isLinearElement(element) && typeof updates.points !== "undefined"` → nope. The `updates` object doesn't contain `type` so it won't work.
- ad `Mutable` → no preference myself, but I was mad it wasn't Typescript' std library, so instead I added it to ours :).